imblearn/base.py

```diff
@@ -7,15 +7,25 @@
 from abc import ABCMeta, abstractmethod
 
 import numpy as np
+import sklearn
 from sklearn.base import BaseEstimator, OneToOneFeatureMixin
 from sklearn.preprocessing import label_binarize
+from sklearn.utils.metaestimators import available_if
 from sklearn.utils.multiclass import check_classification_targets
+from sklearn.utils.fixes import parse_version
 
 from .utils import check_sampling_strategy, check_target_type
+from .utils.fixes import validate_data
 from .utils._param_validation import validate_parameter_constraints
 from .utils._validation import ArraysTransformer
 
 
+def check_version():
+    return parse_version(
+        parse_version(sklearn.__version__).base_version
+    ) >= parse_version("1.6")
```

Comment:
I'd probably do a `try / except` with `import Tags` kinda thing though. With a note that "remove this after 1.6 becomes minimum"

Reply:
This is not the pattern that we document. But actually, I don't know why we are telling to use `available_if`. Right now, we could have both `_more_tags` and `__sklearn_tags__` define.

The `available_if` would make sense if you want to avoid a deprecation warning of an error that we could raise in the future if you define both `_more_tags` and `__sklearn_tags__`. This is something that I mentioned here: https://github.com/scikit-learn/scikit-learn/issues/30257